src/client/pfs_file.go

```diff
@@ -171,61 +174,86 @@ func (mfc *modifyFileCore) maybeError(f func() error) (retErr error) {
 	return f()
 }
 
-func (mfc *modifyFileCore) sendPutFile(req *pfs.PutFile) error {
+func (mfc *modifyFileCore) sendPutFile(req *pfs.AddFile) error {
 	return mfc.client.Send(&pfs.ModifyFileRequest{
-		Modification: &pfs.ModifyFileRequest_PutFile{
-			PutFile: req,
+		Body: &pfs.ModifyFileRequest_AddFile{
+			AddFile: req,
 		},
 	})
 }
 
-func (mfc *modifyFileCore) PutFileTar(r io.Reader, opts ...PutFileOption) error {
+func (mfc *modifyFileCore) PutFileTAR(r io.Reader, opts ...PutFileOption) error {
+	config := &putFileConfig{}
+	for _, opt := range opts {
+		opt(config)
+	}
 	return mfc.maybeError(func() error {
-		pf := &pfs.PutFile{
-			Source: &pfs.PutFile_TarFileSource{
-				TarFileSource: &pfs.TarFileSource{},
-			},
-		}
-		for _, opt := range opts {
-			opt(pf)
-		}
-		if err := mfc.sendPutFile(pf); err != nil {
-			return err
-		}
-		_, err := grpcutil.ChunkReader(r, func(data []byte) error {
-			return mfc.sendPutFile(&pfs.PutFile{
-				Source: &pfs.PutFile_TarFileSource{
-					TarFileSource: &pfs.TarFileSource{
-						Data: data,
-					},
-				},
+		return tarutil.Iterate(r, func(f tarutil.File) error {
```

Comment:
We don't need to use tarutil for this. It is useful when we need to convert to a write, but the standard tar package has good support for tar readers, which is what we need here. Also, as a side note, you need to set the stream flag when using tarutil.Iterate or else it is going to buffer the files in memory (and yes, we should make that a required parameter).